src/common/KokkosKernels_SimpleUtils.hpp

```diff
@@ -260,6 +260,59 @@ bool kk_is_identical_view(view_type1 view1, view_type2 view2, eps_type eps){
   }
 }
 
+template<typename view_type1, typename view_type2, typename eps_type = typename Kokkos::Details::ArithTraits<typename view_type2::non_const_value_type>::mag_type>
+struct IsRelativelyIdenticalFunctor{
+  view_type1 view1;
+  view_type2 view2;
+  eps_type eps;
+
+
+  IsRelativelyIdenticalFunctor(view_type1 view1_, view_type2 view2_, eps_type eps_):
+    view1(view1_), view2(view2_), eps(eps_){}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const size_t &i, size_t &is_equal) const {
+    typedef typename view_type2::non_const_value_type val_type;
+    typedef Kokkos::Details::ArithTraits<val_type> KAT;
+    typedef typename KAT::mag_type mag_type;
+    typedef Kokkos::Details::ArithTraits<mag_type> KATM;
+ 
+    mag_type val_diff = KAT::abs (view1(i) - view2(i));
+    mag_type denominator = KAT::abs(view1(i));
+    if(KAT::abs(view2(i)) > denominator)
+      denominator = KAT::abs(view2(i));
+    
+    if(denominator > KATM::zero() )
+      val_diff = val_diff / denominator;
```

Comment:
That is a strange way to calculate relative diff. I would have done (x1-x2) / norm(norm(x1), norm(x2)). Is this what you are trying to do ?

Reply:
I'm doing:
i) `result = abs(x1 - x2)/ max(abs(x1), abs(x2))` if `max(abs(x1), abs(x2)) > 0`
ii) `result = abs(x1 - x2)` if `max(abs(x1), abs(x2)) = 0`